### ISSUE TITLE
Feature request: Auto-update library

### ISSUE THREAD

**Issue by OOPMan** [cmus/cmus#233]
Hi

Great music player you have here. I would love it if it were able to automatically update the library. I regularly add new music and this is a pretty important feature for me...

**Comment by helveticade** [comment]
+1 to this, watching folders would be really handy

**Comment by OOPMan** [comment]
@flyingmutant Yes, pretty much.

**Comment by elliottetzkorn** [comment]
I would also love this feature!

**Comment by stevenleeg** [comment]
Yeah this would be great. Is there any way to rescan the library's directories at the moment, or do I have to add a new album's folder manually?

**Comment by HarveyHunt** [comment]
I think you can achieve this using an inotify script (or even [when-changed](https://github.com/joh/when-changed)) to pipe a raw command into ```cmus-remote``` to update your library - although I can't remember which command you would need to send to cmus...

**Comment by KabbAmine** [comment]
:+1:  for auto-update functionality

**Comment by bombush** [comment]
+1 ! wached folders would be great

**Comment by DavidOliver** [comment]
Without a watched folder/auto update feature, is the solution to simply re-add a directory? (Asking as a very new cmus user Thanks!)

**Comment by kuzyn** [comment]
@DavidOliver as far as I know, yes: `:clear` and then `:add /your/folder`

**Comment by nefthy** [comment]
While :clear / :add will do the trick. Its not realy elegant.

A :remove <file> command should be quite easy to implement. So an external script that watches a folder could do something like when a file is modified. 
```
:remove /path/to/watched/file.mp3
:add /path/to/watched/file.mp3
```

An implementation in cmus with inotify is not that hard.

@flyingmutant or @mahkoh: can you assign this issue to me. I'll implement that when I find time.

**Comment by dodona2** [comment]
I support that automatically update or rescan the library.

**Comment by XVilka** [comment]
Would be really awesome to have.

**Comment by c0llision** [comment]
Here is how I handle reloading my library.

Create the following script somewhere:
`#!/bin/bash
cmus-remote -C clear
cmus-remote -C "add ~/Music"
cmus-remote -C "update-cache -f"`

Replace ~/Music with the path to your library.

Then do:
`chmod +x your-script.sh`

Then open cmus and run:
` :bind -f common shell u shell ~/path/to/your-script.sh`

Then every time you press 'u', your library will be reloaded. I used the key 'u', because it was already bind to reload-cache, which my script does anyway.

I also tried using the python library watchdog to automatically run the script whenever the library folder changes. However there were some limitations to this.

* Watchdog correctly recognizes when new files or added or existing files are modified, but fails to recognize when files are deleted
* cmus spazzes out when a large number of files are edited (such as when I was using a script to edit my ID3 tags).
* My script will only work if cmus is running when the library is changed.

Nevertheless, if you want to do it this way, you can use [when-changed](https://github.com/joh/when-changed) which is a python program that uses watchdog:

`when-changed ~/Music -c "sh ~/path/to/your-script.sh"`

**Comment by raspbeguy** [comment]
my solution was to execute `:add /path/to/music` on a regular basis. It wouldn't delete missing files but that's not a problem in my case.

**Comment by bettyvschmartz** [comment]
+1 for this. It takes far too long with a large library to add a new album when using `:add /path/to/music`

**Comment by MrAureliusR** [comment]
I find it difficult to believe that this functionality isn't already a part of cmus. I don't think *anyone* has a static music library. Do I really have to go manually add every folder whenever I get a new album ripped?

**Comment by h8uthemost** [comment]
Is there still not a command to refresh your list? Or auto update?

I'm new to cmus, but from what I see, everytime I download a new album I have to hit "5", then hit "a" on whatever folder I store my music in. There's seriously not an auto update?

**Comment by MicahBird** [comment]
While there still isn't an auto update feature, a way to update the library when opening cmus is to add the following to your `~/.config/cmus/rc` file.
```
clear
add ~/Music/
update-cache -f
```
This clears your library, re-adds your Music folder and updates the cache when opening cmus.

**Comment by U1s2e3r4n5a6m7e** [comment]
When I saw an 'update library' thread people always recommend:
```
clear
add music_path_again
```
It drives me crazy (┛ಠ_ಠ)┛彡┻━┻

Why should I add again my **3TB+** music collection, lost my time and lost my old `play_count` counter every time when I wanted a **quick** manual update?
**RTFM!**
I just saying this for **manual-update** not for this(auto-update) request.
It's already implemented:

- `:update-cache` or `u key` : for modification time changed files and deleted files
- `add music_path` or `a key` on Browser(5) view : scan and only add new files. Way faster than _clear and add again_

**Comment by MicahBird** [comment]
My bad! Thanks @U1s2e3r4n5a6m7e! _Scratch my last post..._ The following `~/.config/cmus/rc` would be better:
```
update-cache
add ~/Music
```

**Comment by U1s2e3r4n5a6m7e** [comment]
This is great! @MicahBird 

But other than that misunderstood _manual update_ already infected the whole Reddit posts, Websites, wikis and forums. : (

**Comment by dinotheextinct** [comment]
> 
![image](https://user-images.githubusercontent.com/32158581/213867057-134f94cd-4448-4f46-abc4-dd00019621d0.png)
